psychopy/visual/noise.py

```diff
@@ -442,86 +451,100 @@ def draw(self, win=None):
-            mysf=self.size[0]*self.noiseBaseSf
-            lowsf=self.size[0]*self.noiseFilterLower
-            upsf=self.size[0]*self.noiseFilterUpper
+            mysize = self.texRes
+            pixSize = self.size/self.texRes
+            sampleSize = self.noiseElementSize/pixSize
+            mysf = self.size[0]*self.noiseBaseSf
+            lowsf = self.size[0]*self.noiseFilterLower
+            upsf = self.size[0]*self.noiseFilterUpper
        
-        self._size=mysize  # store for use by updateNoise()
-        self._sf=mysf
+        self._size = mysize  # store for use by updateNoise()
+        self._sf = mysf
         if self.noiseType in ['binary','Binary','normal','Normal','uniform','Uniform']:
-            self._sideLength=numpy.round(mysize/sampleSize)  # dummy side length for use when unpacking noise samples in updateNoise()
+            self._sideLength = numpy.round(mysize/sampleSize)  # dummy side length for use when unpacking noise samples in updateNoise()
             self._sideLength.astype(int)
-            totalSamples=self._sideLength[0]*self._sideLength[1]
+            if ((self._sideLength[0] < 2) and (self._sideLength[1] < 2)):
+                msg=('Noise sample size '
+                     'must result in more than '
+                     '1 sample per image dimension')
+                raise Exception(msg)
+            totalSamples = self._sideLength[0]*self._sideLength[1]
             if self.noiseType in ['binary','Binary']:
                 self.noiseTex=numpy.append(numpy.ones(int(numpy.round(totalSamples/2.0))),-1*numpy.ones(int(numpy.round(totalSamples/2.0))))
         elif self.noiseType in ['White','white']:
-            self.noiseTex=numpy.ones((int(mysize),int(mysize)))
-            self.noiseTex[0][0]=0
+            self.noiseTex = numpy.ones((int(mysize),int(mysize)))
+            self.noiseTex[0][0] = 0
         #elif self.noiseType in ['Coloured','coloured']:
         #    pin=filters.makeRadialMatrix(matrixSize=mysize, center=(0,0), radius=1.0)
         #    self.noiseTex=numpy.multiply(numpy.ones((int(mysize),int(mysize))),(pin)**self.noiseFractalPower) 
         #    self.noiseTex=fftshift(self.noiseTex)
         #    self.noiseTex[0][0]=0
         elif self.noiseType in ['Isotropic','isotropic']:
-            localf=mysf/mysize
-            linbw=2**self.noiseBW
-            lowf=2.0*localf/(linbw+1.0)
-            highf=linbw*lowf
-            FWF=highf-lowf
-            sigmaF=FWF/(2*numpy.sqrt(2*numpy.log(2)))
-            self.noiseTex=numpy.zeros(int(mysize**2))
-            self.noiseTex=numpy.reshape(self.noiseTex,(int(mysize),int(mysize)))
-            pin=filters.makeRadialMatrix(matrixSize=mysize, center=(0,0), radius=2)
-            self.noiseTex=filters.makeGauss(pin, mean=localf, sd=sigmaF)
-            self.noiseTex=fftshift(self.noiseTex)
-            self.noiseTex[0][0]=0
+            if mysf > mysize/2:
+                msg =('Base frequency for isotropic'
```

Comment:
There's a space missing after the `=` and at the end of the line before the closing quotation mark `('... isotropicSPACE')`